### DISCUSSION TITLE
3 base stations & 2 dongles with 4 drones

### DISCUSSION THREAD

**Discussion by Shibaandcorgii** [bitcraze/discussions#81] [Q&A]
Hi !
We want to use 3 base stations to get more accurate of positioning. We follow this document https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/lighthouse/multi_base_stations/  and the third base station doesn’t show on coordinate so we wonder how to make it show on lighthouse positioning tab.

And we also have another question. As we use 1 PA dongle with 4 drones , we notice that they’re unstable . If we use 2 dongles , is the stability will be better ? And how to change the address of PA dongle (0 or 1) ?

Any more essential information please let me know.
Thanks for your support !

**Comment by krichardsson** [comment]
Hi!

> the third base station doesn’t show on coordinate so we wonder how to make it show on lighthouse positioning tab.

The third base station will appear in the client if the Crazyflie has received data from it. The essential steps are:
1. set the channel of the base station to 3 (in your case)
2. increase the number of supported base stations in the Crazyflie firmware. Modify this line https://github.com/bitcraze/crazyflie-firmware/blob/master/src/utils/interface/lighthouse/pulse_processor.h#L40 (make and flash to the Crazyflie)
3. Estimate the system geometry and store it in the Crazyflie using https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/lighthouse/multi_bs_geometry_estimation.py

That should be all that is needed.

Just to double check, you must use Lighthouse V2, Lighthouse V1 only supports 2 base stations.

**Reply by Shibaandcorgii** [reply]
We run multi_bs_geometry_estimation.py and set the third bs as channel 3 , but it’s seem like the position record is from just 2 bs as the picture show.

![94902142-3D24-4296-AE48-3DFAF8CA476E](https://user-images.githubusercontent.com/106365446/173025189-92322978-5d44-41fa-9a76-737518d0555d.jpeg)

Is all of the base station should be visible ?

**Reply by krichardsson** [reply]
Did you do point 2 above and increase the number of supported base stations in the Crazyflie firmware?

**Reply by Shibaandcorgii** [reply]
Yes i did.

**Reply by krichardsson** [reply]
Have you verified that the third base station works? You could change the channel to 1 and replace the first base station just to make sure it is OK

**Reply by Shibaandcorgii** [reply]
Yes , i also used the third bs replaced the first bs too and everything is ok.

**Reply by krichardsson** [reply]
Great, the we know the base station works. To me it sounds as the firmware has not been properly built and flashed. 

The client gets the number of base stations from the Crazyflie through the read only parameter `lighthouse/bsAvailable`, you can check it in the client. This is a bit field, if two base stations are supported the param will be 1+2=3, if three base stations are supported it should be 1+2+4=7. This is what determines how many base stations that show up in the client.

**Comment by krichardsson** [comment]
> We want to use 3 base stations to get more accurate of positioning.

I don't think you will see any dramatic improvements in accuracy, more base stations mainly helps in increasing the covered area. The most important factors for good accuracy are probably:
1. Place the base stations around 90 degrees from each other (not face to face).
2. A good geometry estimation, the script mentioned above should give a much better result than the quick estimation from the client.

**Comment by krichardsson** [comment]
> As we use 1 PA dongle with 4 drones , we notice that they’re unstable . If we use 2 dongles , is the stability will be better ?

In what way are they unstable? If you mean that the position when flying is not stable, it is more likely that it is related to the lighthouse system rather than the Crazyradio. Adding more radios mainly adds more bandwidth if you need to transfer more data. Could you tell us a bit more about your setup and what you are trying to do?

**Comment by krichardsson** [comment]
> And how to change the address of PA dongle (0 or 1) ?

If you use more than one radio, they are simply enumerated as radio:/0/..., radio:/1/... and so on, it is not possible to set the number for a specific radio.

Also if you use more radios, all Crazyflies connected to one radio should use the same channel (frequency) but different addresses, while the radios should use different channels. Something like this:

The first radio might use URIs like this:
`radio://0/30/2M/E7E7E7E701`
`radio://0/30/2M/E7E7E7E702`

and the second radio might use URIs like this:
`radio://1/90/2M/E7E7E7E701`
`radio://1/90/2M/E7E7E7E702`
